src/phase_integrals/phase_integrals.jl

```diff
@@ -0,0 +1,119 @@
+@inline function _field_squared_integral(
+    field::AbstractPulsedPlaneWaveField,
+    pol::AbstractPolarization,
+    phi::T
+) where {T<:Real}
+    return quadgk(t -> amplitude(field, pol, t)^2, 0.0, phi)[1]
+end
+
+# kinematic vector factor alpha_1^mu appearing in Volkov phase
+# TODO: Is `ELEMENTARY_CHARGE` actually related to the scattering particle or just a factor
+@inline function _kinematic_vector_phase_factor(
+    field::AbstractPulsedPlaneWaveField,
+    p_in::MT,
+    p_out::MT,
+) where {MT<:AbstractFourMomentum}
+    k_mu = momentum(field)
+    return ELEMENTARY_CHARGE*(p_out/(k_mu*p_out) - p_in/(k_mu*p_in))
+end
+
+# kinematic scalar factor alpha_2 appearing in Volkov phase
+# TODO: Is `ELEMENTARY_CHARGE` actually related to the scattering particle or just a factor
+@inline function _kinematic_scalar_phase_factor(
+    field::AbstractPulsedPlaneWaveField,
+    p_in::MT,
+    p_out::MT,
+) where {MT<:AbstractFourMomentum}
+    k_mu = momentum(field)
+    return ELEMENTARY_CHARGE_SQUARE * (1/(k_mu*p_in) - 1/(k_mu*p_out))
+end
+
+# non-linear Volkov phase
+"""
+
+    _phase_function(field::AbstractPulsedPlaneWaveField, pol::AbstractPolarization, p_in::, p_out::, phi::)
+
+Return the phase function (or non-linear Volkov phase), for the given phase space point `p_in, p_out` and a given phase value `phi`.
+
+!!! note "Convention"
+
+    The non-linear Volkov phase is defined as:
+
+    ```math
+    \\begin{align*}
+        G(\\varphi,p, p^\\prime)& = \\alpha_1^\\mu \\int\\limits_0^\\varphi \\mathrm{d}\\varphi^\\prime A_\\mu(\\varphi^\\prime)
+            + \\alpha_2 \\int\\limits_0^\\varphi \\mathrm{d}\\varphi^\\prime A^2(\\varphi^\\prime) \\\\
+    \\end{align*}
+    ```
+    where ``A^\\mu(\\varphi)`` is the background field, ``\\alpha_1^\\mu`` is the [`kinematic vector phase factor`](@ref), and ``\\alpha_2`` is the [`kinematic scalar phase factor`](@ref).
+    Both ``\\alpha_1^\\mu`` and ``\\alpha_2`` depend on the given phase space point ``(p,p^\\prime)`` and the field's reference momentum ``k^\\mu`` the photon number parameter.
+"""
+@inline function _phase_function(
+    field::AbstractPulsedPlaneWaveField,
+    pol::AbstractPolarization,
+    p_in::MT,
+    p_out::MT,
+    phi::T
+) where {MT<:AbstractFourMomentum, T<:Real}
+    first = _kinematic_vector_phase_factor(field, p_in, p_out) * _field_integral(field, pol, phi)
+    second = _kinematic_scalar_phase_factor(field, p_in, p_out) * _field_squared_integral(field, pol, phi)
+    return first+second
+end
+
+# integrand shared by all phase integrals
+@inline function _shared_integrand(
+    field::AbstractPulsedPlaneWaveField,
+    pol::AbstractPolarization,
+    p_in::MT,
+    p_out::MT,
+    phi::T,
+    pnum::T
+) where {MT<:AbstractFourMomentum, T<:Real}
+    return exp(1im*(pnum*phi + _phase_function(field, pol, p_in, p_out, phi)))
+end
+
+
+include("zeroth_integral.jl")
+include("first_integral.jl")
+include("second_integral.jl")
```

Comment:
Per the coding guidelines, we only use includes at the top-level, so these includes should be moved to the QEDfields.jl file.